### ISSUE TITLE
Current source code doesn't compile correctly

### ISSUE THREAD

**Issue by MrSpock** [dxps/fullstack-rust-axum-dioxus-rwa#21]
When trying to run example (git clone..; ./run_dev.sh) I'm getting following error:
```$./run_dev.sh
[INFO] 🚅 Running build command...
[ERROR] 🚫 Serve startup failed: error[E0107]: this struct takes 2 generic arguments but 1 generic argument was supplied
   --> /home/spock/.cargo/registry/src/github.com-1ecc6299db9ec823/dioxus-core-0.2.1/src/virtual_dom.rs:109:19
    |
109 |     dirty_scopes: IndexSet<ScopeId>,
    |                   ^^^^^^^^ ------- supplied 1 generic argument
    |                   |
    |                   expected 2 generic arguments
    |
note: struct defined here, with 2 generic parameters: `T`, `S`
   --> /home/spock/.cargo/registry/src/github.com-1ecc6299db9ec823/indexmap-1.9.3/src/set.rs:67:12
    |
67  | pub struct IndexSet<T, S> {
    |            ^^^^^^^^ -  -
help: add missing generic argument
    |
109 |     dirty_scopes: IndexSet<ScopeId, S>,
    |                                   +++```

Looks like libraries used by project has some issues.

**Comment by MrSpock** [comment]
Error exactly the same like here:
https://github.com/OpenAPITools/openapi-generator/issues/12154

Adding indexmap = {version = "1.8.1", features = ["std"]}  to dioxus-core Cargo.toml solves problem.

**Comment by dxps** [comment]
Interesting. I just tried again from scratch and it's all good.
Using either rust 1.68.0 or the currently available 1.68.2.

```shell
…/backend on  main ❯ rm -rf ../target/
…/backend on  main ❯ 
```

```shell
…/backend on  main ❯ ./run_dev.sh 
[Running 'cargo run --bin server -- --port 8081 --log debug']
   Compiling libc v0.2.132
   Compiling cfg-if v1.0.0
   Compiling autocfg v1.1.0
   Compiling proc-macro2 v1.0.50
   Compiling quote v1.0.21
   Compiling unicode-ident v1.0.3
   Compiling syn v1.0.107
   Compiling version_check v0.9.4
   Compiling once_cell v1.13.1
   Compiling log v0.4.17
   Compiling memchr v2.5.0
   Compiling smallvec v1.9.0
   Compiling bytes v1.2.1
   Compiling scopeguard v1.1.0
   Compiling lock_api v0.4.7
   Compiling pin-project-lite v0.2.9
   Compiling typenum v1.15.0
   Compiling generic-array v0.14.6
   Compiling futures-core v0.3.23
   Compiling itoa v1.0.3
   Compiling serde_derive v1.0.152
   Compiling parking_lot_core v0.9.3
   Compiling serde v1.0.152
   Compiling getrandom v0.2.7
   Compiling tokio v1.24.2
   Compiling signal-hook-registry v1.4.0
   Compiling parking_lot v0.12.1
   Compiling socket2 v0.4.4
   Compiling mio v0.8.4
   Compiling num_cpus v1.13.1
   Compiling cc v1.0.73
   Compiling subtle v2.4.1
   Compiling slab v0.4.7
   Compiling futures-sink v0.3.23
   Compiling futures-task v0.3.23
   Compiling num-traits v0.2.15
   Compiling futures-util v0.3.23
   Compiling spin v0.5.2
   Compiling crypto-common v0.1.6
   Compiling block-buffer v0.10.2
   Compiling untrusted v0.7.1
   Compiling digest v0.10.3
   Compiling minimal-lexical v0.2.1
   Compiling pin-utils v0.1.0
   Compiling base64 v0.13.0
   Compiling num-integer v0.1.45
   Compiling ahash v0.7.6
   Compiling nom v7.1.1
   Compiling fnv v1.0.7
   Compiling rustversion v1.0.11
   Compiling cpufeatures v0.2.4
   Compiling percent-encoding v2.1.0
   Compiling matches v0.1.9
   Compiling http v0.2.8
   Compiling unicase v2.6.0
   Compiling ring v0.16.20
   Compiling tracing-core v0.1.30
   Compiling unicode-segmentation v1.9.0
   Compiling bitflags v1.3.2
   Compiling serde_json v1.0.91
   Compiling ryu v1.0.11
   Compiling futures-channel v0.3.23
   Compiling tinyvec_macros v0.1.0
   Compiling tinyvec v1.6.0
   Compiling heck v0.4.0
   Compiling form_urlencoded v1.0.1
   Compiling thiserror v1.0.38
   Compiling rustls v0.20.6
   Compiling crc32fast v1.3.2
   Compiling parking_lot_core v0.8.5
   Compiling alloc-no-stdlib v2.0.3
   Compiling alloc-stdlib v0.2.1
   Compiling unicode-normalization v0.1.21
   Compiling hashbrown v0.12.3
   Compiling indexmap v1.9.1
   Compiling instant v0.1.12
   Compiling crossbeam-utils v0.8.11
   Compiling async-trait v0.1.57
   Compiling unicode-bidi v0.3.8
   Compiling tower-service v0.3.2
   Compiling mime v0.3.16
   Compiling adler v1.0.2
   Compiling mime_guess v2.0.4
   Compiling miniz_oxide v0.5.3
   Compiling brotli-decompressor v2.3.2
   Compiling http-body v0.4.5
   Compiling rand_core v0.6.3
   Compiling dirs-sys v0.3.7
   Compiling proc-macro-error-attr v1.0.4
   Compiling ppv-lite86 v0.2.16
   Compiling httparse v1.7.1
   Compiling httpdate v1.0.2
   Compiling crossbeam-queue v0.3.6
   Compiling tower-layer v0.3.2
   Compiling either v1.8.0
   Compiling rand_chacha v0.3.1
   Compiling dirs v4.0.0
   Compiling webpki v0.22.0
   Compiling sct v0.7.0
   Compiling itertools v0.10.3
   Compiling brotli v3.3.4
   Compiling flate2 v1.0.24
   Compiling parking_lot v0.11.2
   Compiling idna v0.2.3
   Compiling axum-core v0.3.1
   Compiling hmac v0.12.1
   Compiling time v0.1.44
   Compiling num-bigint v0.4.3
   Compiling proc-macro-error v1.0.4
   Compiling unicode_categories v0.1.1
   Compiling lazy_static v1.4.0
   Compiling iana-time-zone v0.1.46
   Compiling try-lock v0.2.3
   Compiling tokio-macros v1.8.0
   Compiling thiserror-impl v1.0.38
   Compiling tracing-attributes v0.1.23
   Compiling pin-project-internal v1.0.12
   Compiling crc-catalog v2.1.0
   Compiling crc v3.0.0
   Compiling pin-project v1.0.12
   Compiling want v0.3.0
   Compiling sqlformat v0.2.1
   Compiling url v2.2.2
   Compiling hkdf v0.12.3
   Compiling futures-intrusive v0.4.0
   Compiling tracing v0.1.37
   Compiling webpki-roots v0.22.4
   Compiling rand v0.8.5
   Compiling dotenvy v0.15.2
   Compiling stringprep v0.1.2
   Compiling hashlink v0.8.0
   Compiling axum v0.6.2
   Compiling iri-string v0.4.1
   Compiling headers-core v0.2.0
   Compiling atoi v1.0.0
   Compiling sha2 v0.10.2
   Compiling sha1 v0.10.4
   Compiling sha-1 v0.10.0
   Compiling rustls-pemfile v1.0.1
   Compiling md-5 v0.10.1
   Compiling paste v1.0.8
   Compiling byteorder v1.4.3
   Compiling time-macros v0.2.4
   Compiling num_threads v0.1.6
   Compiling event-listener v2.5.3
   Compiling hex v0.4.3
   Compiling tokio-rustls v0.23.4
   Compiling tower v0.4.13
   Compiling sqlx-rt v0.6.2
   Compiling tokio-stream v0.1.9
   Compiling async-compression v0.3.14
   Compiling tokio-util v0.7.3
   Compiling whoami v1.2.1
   Compiling http-range-header v0.3.0
   Compiling hyper v0.14.20
   Compiling headers v0.3.7
   Compiling axum-macros v0.3.1
   Compiling time v0.3.14
   Compiling sync_wrapper v0.1.1
   Compiling matchit v0.7.0
   Compiling os_str_bytes v6.3.0
   Compiling overload v0.1.1
   Compiling linked-hash-map v0.5.6
   Compiling nu-ansi-term v0.46.0
   Compiling yaml-rust v0.4.5
   Compiling simple_asn1 v0.6.2
   Compiling uuid v1.2.2
   Compiling chrono v0.4.23
   Compiling serde_path_to_error v0.1.9
   Compiling tower-http v0.3.5
   Compiling serde_urlencoded v0.7.1
   Compiling clap_lex v0.2.4
   Compiling clap_derive v3.2.17
   Compiling sharded-slab v0.1.4
   Compiling tracing-log v0.1.3
   Compiling pem v1.1.0
   Compiling sqlx-core v0.6.2
   Compiling atty v0.2.14
   Compiling thread_local v1.1.4
   Compiling textwrap v0.15.0
   Compiling termcolor v1.1.3
   Compiling pathdiff v0.2.1
   Compiling strsim v0.10.0
   Compiling zeroize v1.5.7
   Compiling deunicode v0.4.3
   Compiling secrecy v0.8.0
   Compiling config v0.13.3
   Compiling tracing-subscriber v0.3.16
   Compiling slug v0.1.4
   Compiling jsonwebtoken v8.2.0
   Compiling fastrand v1.8.0
   Compiling md5 v0.7.0
   Compiling clap v3.2.17
   Compiling axum-extra v0.4.2
   Compiling sqlx-macros v0.6.2
   Compiling sqlx v0.6.2
   Compiling backend v0.1.0 (/home/dxps/dev/dxps_gh/fullstack-rust-axum-dioxus-rwa/backend)
    Finished dev [unoptimized + debuginfo] target(s) in 1m 10s
     Running `/home/dxps/dev/dxps_gh/fullstack-rust-axum-dioxus-rwa/target/debug/server --port 8081 --log debug`
2023-03-28T15:49:02.479154Z  INFO server: Connected to the database (with 2 conns).    
2023-03-28T15:49:02.480225Z  INFO server: Listening for requests on http://[::1]:8081 ...
```

**Comment by dxps** [comment]
Without further activity on this aspect, I'll close the issue.
